src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -785,11 +785,10 @@ protected BoundCall MakeQueryInvocation(CSharpSyntaxNode node, BoundExpression r
 
         protected BoundExpression MakeConstruction(CSharpSyntaxNode node, NamedTypeSymbol toCreate, ImmutableArray<BoundExpression> args, DiagnosticBag diagnostics)
         {
-            AnalyzedArguments analyzedArguments = AnalyzedArguments.GetInstance();
-            analyzedArguments.Arguments.AddRange(args);
-            var result = BindClassCreationExpression(node, toCreate.Name, node, toCreate, analyzedArguments, diagnostics);
+            Debug.Assert(toCreate.IsAnonymousType);
+
+            var result = new BoundAnonymousObjectCreationExpression(node, toCreate.InstanceConstructors[0], args, ImmutableArray<BoundAnonymousPropertyDeclaration>.Empty, toCreate);
```

Comment:
This is potentially cutting out a good deal of error checking done in BindClassCreationExpression. Are we sure this is the right change to make?

Reply:
I was hoping unit tests to catch any missing functionality, if it was indeed desired or possible to hit those cases. No C# compiler unit tests failed locally for me after this change.

Reply:
Adding @gafter @jcouv to confirm.

Reply:
>  No C# compiler unit tests failed locally for me after this change.

That is not good evidence of a change being correct when the change itself is a bit speculative. It's more likely to indicate there is a test gap.

Reply:
Filed https://github.com/dotnet/roslyn/issues/28157 for the compiler team to make the correct change and unit tests at a later milestone. I will update this PR to put a workaround in operation factory as suggested by @AlekseyTs